cmd/vic-machine/common/cnetworks.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"encoding"
+	"fmt"
+	"net"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"gopkg.in/urfave/cli.v1"
+
+	"github.com/vmware/vic/pkg/ip"
+)
+
+// Container networks - mapped from vSphere
+type CNetworks struct {
+	ContainerNetworks         cli.StringSlice `arg:"container-network"`
+	ContainerNetworksGateway  cli.StringSlice `arg:"container-network-gateway"`
+	ContainerNetworksIPRanges cli.StringSlice `arg:"container-network-ip-range"`
+	ContainerNetworksDNS      cli.StringSlice `arg:"container-network-dns"`
+	IsSet                     bool
+}
+
+// CNetworkFields holds items returned by ProcessContainerNetworks for each
+// container network.
+type CNetworkFields struct {
+	VICNet   string
+	VNet     string
+	Gateways net.IPNet
+	IPRanges []ip.Range
+	DNS      []net.IP
+}
+
+func (c *CNetworks) CNetworkFlags(hidden bool) []cli.Flag {
+	return []cli.Flag{
+		cli.StringSliceFlag{
+			Name:  "container-network, cn",
+			Value: &c.ContainerNetworks,
+			Usage: "vSphere network list that containers can use directly with labels, e.g. vsphere-net:backend. Defaults to DCHP - see advanced help (-x).",
+		},
+		cli.StringSliceFlag{
+			Name:   "container-network-gateway, cng",
+			Value:  &c.ContainerNetworksGateway,
+			Usage:  "Gateway for the container network's subnet in CONTAINER-NETWORK:SUBNET format, e.g. vsphere-net:172.16.0.1/16",
+			Hidden: hidden,
```

Comment:
the advanced container network options are all hidden in vic-machine create, because we have '-x' option to show all advanced options, but in configure, we're not supporting all create options at this time, it might be better to keep all of them open atm.
You can allow the method to accept one option, to make it different in create and configure command